components/script/dom/shadowroot.rs

```diff
@@ -225,7 +242,7 @@ impl ShadowRootMethods for ShadowRoot {
 
 #[allow(unsafe_code)]
 pub trait LayoutShadowRootHelpers {
-    unsafe fn get_host_for_layout(&self) -> LayoutDom<Element>;
+    unsafe fn get_host_for_layout(&self) -> Option<LayoutDom<Element>>;
```

Comment:
Probably this should just keep as-is and the function unwrap it, since we should never do layout on a detached shadow root.